builder/hyperv/common/step_create_build_dir.go

```diff
@@ -62,6 +65,12 @@ func (s *StepCreateBuildDir) Cleanup(state multistep.StateBag) {
 	}
 
 	ui := state.Get("ui").(packersdk.Ui)
+
+	if s.KeepRegistered {
```

Comment:
I would suggest checking if the build succeeded before keeping the artifacts, we should probably remove it if the build failed or was aborted for any reason.

```suggestion
	_, cancelled := state.GetOk(multistep.StateCancelled)
	_, halted := state.GetOk(multistep.StateHalted)

	if s.KeepRegistered && !(cancelled || halted) {
```